ramips: add missing "pinctrl-names" for Youku YK1

DragonBluep opened the pull request:
Without this definition ethernet led can work as usual, but it's better to
re-add it. Relying on default values may cause uncontrollable factors.

Fixes: 882a611 ("ramips: improve pinctrl for Youku YK-L1")

adschm commented:
Please use the following to indicate the fix:

`Fixes: 882a6116d3d6 ("ramips: improve pinctrl for Youku YK-L1")`

... instead of your "fix commit ..."

DragonBluep commented:
Thanks for your reply,  I have fixed it.

> Please use the following to indicate the fix in the commit message:
> 
> `Fixes: 882a6116d3d6 ("ramips: improve pinctrl for Youku YK-L1")`
> 
> ... instead of your "fix commit ..."

mpratt14 commented:
I am curious why this is necessary

According to MT7620 programming guide, EPHY led pins are set this way by default (not GPIO mode by default)
the commit you're referencing also states "led "wan" can be triggered by ethernet pinctrl by default"

what happens if you remove both lines? so ethernet block would just be:

```
&ethernet {
	mtd-mac-address = <&factory 0x4>;

	mediatek,portmap = "llllw";
};
```

DragonBluep commented:
> I am curious why this is necessary
> 
> According to MT7620 programming guide, EPHY led pins are set this way by default (not GPIO mode by default)
> the commit you're referencing also states "led "wan" can be triggered by ethernet pinctrl by default"
> 
> what happens if you remove both lines? so ethernet block would just be:
> 
> ```
> &ethernet {
> 	mtd-mac-address = <&factory 0x4>;
> 
> 	mediatek,portmap = "llllw";
> };
> ```

After testing, when ephy pinctrl definition be dropped, the ethernet leds can work well as usual. And I also found there are many similar "useless" pinctrl definitions in other dts files. Now I really wonder why they exist.

mpratt14 commented:
so the line `pinctrl-names` should be removed? and then it works?

DragonBluep commented:
> so the line `pinctrl-names` should be removed? and then it works?


Yes, LEDs can be successfully triggered. I think it can be removed but I am not a Linux expert so I can't decide.
a45cde91183753d5e909c31f07370c5755113439 (ramips: ZTE-Q7: fix switch driver initialization in DTS (FS#716))
Jo-Philipp Wich said the `pinctrl-names` is obsolete here.
I will test another device later.

mpratt14 commented:
> I think it can be removed but I am not a Linux expert so I can't decide.

I am also not a linux expert so I cannot decide either, that is why we need someone like you who is able to test 😄

mpratt14 commented:
in the commit you referenced.....I am not sure what JOW means by "obsoleted by the changes"

I think he is referring to another set of changes....maybe in the DTSI

the 2 changes in that commit are not related

DragonBluep commented:
> I am curious why this is necessary
> 
> According to MT7620 programming guide, EPHY led pins are set this way by default (not GPIO mode by default)
> the commit you're referencing also states "led "wan" can be triggered by ethernet pinctrl by default"
> 
> what happens if you remove both lines? so ethernet block would just be:
> 
> ```
> &ethernet   @#{
> 	mtd-mac-address = <&factory 0x4>;
> 
> 	mediatek,portmap = "llllw";
> };
> ```

I tested this change in Youku YK-L1, DomyWifi DM203 #4054 and Haier HW-L1W #4050, It seems that these definitions are redundant:
```
	pinctrl-names = "default";
	pinctrl-0 = <&ephy_pins>;
```
Without these definitions ethernet leds can work well as usual. I found there are still some devices use these definitions. Maybe we need an additional patch to remove these definitions all at once. And this pull request can be closed.

981213 commented:
@mpratt14 

> According to MT7620 programming guide, EPHY led pins are set this way by default (not GPIO mode by default)
> the commit you're referencing also states "led "wan" can be triggered by ethernet pinctrl by default"

We should not rely on default values. Everything should be explicitly defined if possible, otherwise changes made by bootloader could break it.

@DragonBluep 

> Without these definitions ethernet leds can work well as usual. I found there are still some devices use these definitions. Maybe we need an additional patch to remove these definitions all at once.

NACK on that.

mpratt14 commented:
of course, each device would need to be tested to make sure, but I think it is likely that most boards with these properties defined do not need them

981213 commented:
@mpratt14
> of course, each device would need to be tested to make sure, but I think it is likely that most boards with these properties defined do not need them

We should never rely on default values because there's no guarentees that these values won't be changed by a custom bootloader. We don't want our firmware to be affected by those uncontrollable outside components, and it has nothing to do with testing.

The removal of pinctrl in a45cde9 is probably a misunderstanding of that property: "ephy" is confusingly used as pinctrl for ethernet phy LEDs instead of ethernet phy itself. (It doesn't matter there because ZTE Q7 has no ethernet LED at all. It should be removed, but the reason given in the commit message is wrong.)

DragonBluep committed:
ramips: add missing "pinctrl-names" for Youku YK1

Without this definition ethernet led can work as usual, but it's better to
re-add it. Relying on default values may cause uncontrollable factors.

Fixes: 882a6116d3d6 ("ramips: improve pinctrl for Youku YK-L1")

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep commented:
Thanks for your explanation, I re-added it. @981213

mpratt14 commented:
@981213 



> The removal of pinctrl in [a45cde9](https://github.com/openwrt/openwrt/commit/a45cde91183753d5e909c31f07370c5755113439) is probably a misunderstanding of that property

this explanation makes sense, as the 2 changes are not related to each other, but the commit message does not say that...



> We should never rely on default values because there's no guarantees that these values won't be changed by a custom bootloader.

I agree with all your points here. Maybe there is some way to put these lines in the DTSI without affecting boards that use EPHY_LEDS as a GPIO? (It is about 30 boards that have ephy set to GPIO....)

For boards that have GMAC interface instead, it is already not an issue, because the same properties are used for the MDIO and RGMII pins, so downstream DTS would overwrite them.

Otherwise, it is better to handle one device at a time...

981213 commented:
Merged into my staging tree at: https://git.openwrt.org/?p=openwrt/staging/981213.git
Thanks!
